### DISCUSSION TITLE
Add a new test case for Includes.

### DISCUSSION THREAD

**Discussion by huanggm** [type-challenges/type-challenges#1432] [Q&A]
Test Case: Expect<Equal<Includes<[boolean, 2, 3, 5, 6, 7], false>, false>>

And this answer cannot work properly.

type Includes<T extends readonly any[], U> = U extends T[number] ? true : false;

Is there a more precise answer?

**Comment by Yash-Singh1** [comment]
This works (#1568):

```ts
type Includes<T extends readonly any[], U> = {
  [P in T[number]]: true
}[U] extends true ? true : false;
```

However, it doesn't support anything other than `PropertyKey` in `Item`.

**Reply by ensconced** [reply]
Is there a solution that does work with more than just `PropertyKey`? Currently there is this test case:

```ts
Expect<Equal<Includes<[{}], { a: 'A' }>, false>>
```

But it's a bit misleading because with the solution given above it will fail if you try this:

```ts
Expect<Equal<Includes<[{ a: 'A' }], { a: 'A' }>, true>> // error
```

**Reply by Yash-Singh1** [reply]
I created a PR in a typescript utility package a few weeks ago: https://github.com/sindresorhus/type-fest/blob/main/source/includes.d.ts#L26-L31
